src/python/zquantum/optimizers/search_points_optimizer.py

```diff
@@ -0,0 +1,65 @@
+from zquantum.core.history.recorder import recorder as _recorder
+from zquantum.core.interfaces.functions import CallableWithGradient
+from zquantum.core.interfaces.optimizer import (
+    Optimizer,
+    optimization_result,
+    construct_history_info,
+)
+from zquantum.core.typing import RecorderFactory
+
+from scipy.optimize import OptimizeResult
+from typing import Optional, List
+import numpy as np
+
+
+class SearchPointsOptimizer(Optimizer):
+    def __init__(
+        self,
+        parameter_values_list: List[np.ndarray],
+        recorder: RecorderFactory = _recorder,
+    ):
+        """
+        Args:
+            parameter_values_list: list of parameter values to evaluate
+            recorder: recorder object which defines how to store the optimization history.
+        """
+        super().__init__(recorder=recorder)
```

Comment:
What about adding an assertion here that `len(parameter_values_list)>0`?
And appropriate test of course :)

Reply:
Oh good point!

Reply:
Addressed here: d1556cd